lambda/shared/src/test/scala/feral/lambda/events/ApplicationLoadBalancerEventSuite.scala

```diff
@@ -0,0 +1,193 @@
+import munit.FunSuite
+import scodec.bits.ByteVector
+
+object ApplicationLoadBalancerEventSuite {
+  def allFieldsEvent = json"""
+    {
+      "requestContext": {
+        "elb": {
+          "targetGroupArn": "arn:aws:elasticloadbalancing:us-west-2:123456789012:targetgroup/my-target-group/6d0ecf831eec9f09"
+        }
+      },
+      "httpMethod": "GET",
+      "path": "/lambda",
+      "queryStringParameters": {
+        "query": "1234ABCD"
+      },
+      "headers": {
+        "accept": "text/html,application/xhtml+xml",
+        "accept-language": "en-US,en;q=0.8",
+        "content-type": "text/plain",
+        "cookie": "cookies",
+        "host": "lambda-846800462-us-west-2.elb.amazonaws.com",
+        "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6)",
+        "x-amzn-trace-id": "Root=1-58337364-23a8c76965a2ef7629b2b5c2",
+        "x-forwarded-for": "72.21.198.64",
+        "x-forwarded-port": "443",
+        "x-forwarded-proto": "https"
+      },
+      "multiValueQueryStringParameters": null,
+      "multiValueHeaders": null,
+      "body": null,
+      "isBase64Encoded": false
+    }
+  """
+
+  def withBodyEvent = {
+    val encodedBody =
+      java.util.Base64.getEncoder.encodeToString("hello world".getBytes("UTF-8"))
+    Json.obj(
+      "requestContext" -> Json.obj(
+        "elb" -> Json.obj(
+          "targetGroupArn" -> Json.fromString(
+            "arn:aws:elasticloadbalancing:us-west-2:123456789012:targetgroup/my-target-group/6d0ecf831eec9f09")
+        )
+      ),
+      "httpMethod" -> Json.fromString("POST"),
+      "path" -> Json.fromString("/submit"),
+      "queryStringParameters" -> Json.Null,
+      "headers" -> Json.Null,
+      "multiValueQueryStringParameters" -> Json.Null,
+      "multiValueHeaders" -> Json.Null,
+      "body" -> Json.fromString(encodedBody),
+      "isBase64Encoded" -> Json.fromBoolean(true)
+    )
+  }
+
+  def missingOptionalsEvent = json"""
+    {
+      "requestContext": {
+        "elb": {
+          "targetGroupArn": "arn:aws:elasticloadbalancing:us-west-2:123456789012:targetgroup/my-target-group/6d0ecf831eec9f09"
+        }
+      },
+      "httpMethod": "GET",
+      "path": "/only-required",
+      "isBase64Encoded": false
+    }
+  """
+}
+
+class ApplicationLoadBalancerEventSuite extends FunSuite {
+  import ApplicationLoadBalancerEventSuite.*
+
+  test("decode with all fields populated") {
+    val decoded = allFieldsEvent.as[ApplicationLoadBalancerRequestEvent].toTry.get
+    val expected = ApplicationLoadBalancerRequestEvent(
+      requestContext = ApplicationLoadBalancerRequestContext(
+        elb = Elb(
+          "arn:aws:elasticloadbalancing:us-west-2:123456789012:targetgroup/my-target-group/6d0ecf831eec9f09")
+      ),
+      httpMethod = "GET",
+      path = "/lambda",
+      queryStringParameters = Some(Map("query" -> "1234ABCD")),
+      headers = Some(
+        Map(
+          org.typelevel.ci.CIString("accept") -> "text/html,application/xhtml+xml",
```

Comment:
There is a special syntax to shorten this.

```suggestion
          ci"accept" -> "text/html,application/xhtml+xml",
```